src/ProductConstructionService/ProductConstructionService.ReproTool/ReproTool.cs

```diff
@@ -0,0 +1,280 @@
+        {
+            throw new ArgumentException($"Code flow subscription incorrectly configured: is missing SourceDirectory or TargetDirectory");
+        }
+
+        await darcProcessManager.InitializeAsync();
+
+        var defaultChannel = (await barClient.GetDefaultChannelsAsync(repository: subscription.SourceRepository, channel: subscription.Channel.Name)).First();
+
+        string vmrBranch, productRepoUri, productRepoBranch;
+        bool isForwardFlow;
+        // Check if it's forward flow
+        if (!string.IsNullOrEmpty(subscription.TargetDirectory))
+        {
+            vmrBranch = subscription.TargetBranch;
+            productRepoUri = subscription.SourceRepository;
+            productRepoBranch = defaultChannel.Branch;
+            isForwardFlow = true;
+        }
+        // Backward flow
+        else
+        {
+            vmrBranch = defaultChannel.Branch;
+            productRepoUri = subscription.TargetRepository;
+            productRepoBranch = subscription.TargetBranch;
+            isForwardFlow = false;
+        }
+        var productRepoForkUri = ProductRepoFormat + productRepoUri.Split('/').Last();
+
+        await using var vmrTmpBranch = await PrepareVmrForkAsync(vmrBranch, productRepoUri, productRepoForkUri);
+
+        await PrepareProductRepoForkAsync(productRepoUri, productRepoForkUri, productRepoBranch);
+
+        // Find the latest commit in the source repo to create a build from
+        string sourceRepoSha;
+        (string sourceRepoOwner, string sourceRepoName) = GetRepoOwnerAndName(subscription.SourceRepository);
+        if (string.IsNullOrEmpty(options.Commit))
+        {
+            var res = await ghClient.Git.Reference.Get(sourceRepoOwner, sourceRepoName, $"heads/{defaultChannel.Branch}");
+            sourceRepoSha = res.Object.Sha;
+        }
+        else
+        {
+            // Validate that the commit actually exists
+            try
+            {
+                await ghClient.Repository.Commit.Get(sourceRepoOwner, sourceRepoName, options.Commit);
+            }
+            catch (NotFoundException)
+            {
+                throw new ArgumentException($"Commit {options.Commit} doesn't exist in repo {subscription.SourceRepository}");
+            }
+            sourceRepoSha = options.Commit;
+        }
+
+        // Create the temp channel
+        var channelName = Guid.NewGuid().ToString();
+        await using var channel = await darcProcessManager.CreateTestChannelAsync(channelName);
+
+        // The build
+        var build = await CreateBuildAsync(
+            isForwardFlow ? productRepoForkUri : VmrForkUri,
+            isForwardFlow ? defaultChannel.Branch : vmrTmpBranch.Value,
+            sourceRepoSha);
+
+        // And the subscription
+        await using var testSubscription = await darcProcessManager.CreateSubscriptionAsync(
+            channel: channelName,
```

Comment:
I think that adding comments "Create X" over lines containing `CreateX()` method calls are quite useless.

```suggestion
        var channelName = Guid.NewGuid().ToString();
        await using var channel = await darcProcessManager.CreateTestChannelAsync(channelName);

        var build = await CreateBuildAsync(
            isForwardFlow ? productRepoForkUri : VmrForkUri,
            isForwardFlow ? defaultChannel.Branch : vmrTmpBranch.Value,
            sourceRepoSha);

        await using var testSubscription = await darcProcessManager.CreateSubscriptionAsync(
            channel: channelName,
```